dvc/remote/s3.py

```diff
@@ -279,7 +286,7 @@ def _generate_download_url(self, path_info, expires=3600):
         )
 
     def walk_files(self, path_info, max_items=None):
-        for fname in self._list_paths(path_info, max_items):
+        for fname in self._list_paths(path_info / "", max_items):
```

Comment:
Shouldn't walk_files() work for non-dirs too?

Reply:
`walk_files` is for walking the tree, and we emulate hierarchy via `/` (forward-slash) delimiter. So, I don't think, it should support non-dirs. Any uses?

Reply:
@Suor indeed, walk_files is indeed not intended to work with files.